runtime/lua/vim/lsp/inlay_hint.lua

```diff
@@ -98,6 +98,91 @@ function M.on_refresh(err, _, ctx, _)
   return vim.NIL
 end
 
+--- @class InlayHintOptions
+--- @field bufnr number|nil
```

Comment:
Nit: I think using `?` instead of `|nil` is preferred.

Reply:
But either works with our tooling, which is the important bit.

Reply:
Changed to `?` (in TypeScript this would be the only correct way b/c of assigning `undefined`; in Lua it's just for aesthetics)